### PR TITLE
Raftstore-v2: update peer state after persisting snapshot

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #12842

What's Changed:

```commit-message
Update peer state after persisting snapshot
```
If peer 1 is added to store 1 before the store1 starts, peer 1 will be in learner state. So after applying the snapshot, we should align the state with region's relevant peer state.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```

### CODE REVIEW

**Review by tabokie — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Review by iosmanthus — APPROVED** [review]

File: components/raftstore-v2/src/operation/ready/snapshot.rs [hunk 1]
```diff
@@ -259,6 +259,10 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 meta.region_read_progress
                     .insert(region_id, self.read_progress().clone());
             }
+
+            let region_state = self.raft_group().store().region_state().clone();
+            self.storage_mut().set_region_state(region_state);
```
Comment (BusyJay) [inline components/raftstore-v2/src/operation/ready/snapshot.rs:264]: The state should be set in `apply_snapshot`.

Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/ready/snapshot.rs:264]: Fine, I will propose another PR for it.

**Pull request merged**